### ISSUE TITLE
Authentication failure after inserting new key with Vagrant 1.8.5 (bsd, not linux)

### ISSUE THREAD

**Issue by adfernandes** [mitchellh/vagrant#7727]
This is a note that should be linked to #7610 which has been locked.

The same bug, I believe, is in `/plugins/guests/bsd/cap/public_key.rb`. (BSD vs Linux)

So the patch would be:
```
...
             if test -f ~/.ssh/authorized_keys; then
               grep -v -x -f '#{remote_path}' ~/.ssh/authorized_keys > ~/.ssh/authorized_keys.tmp
               mv ~/.ssh/authorized_keys.tmp ~/.ssh/authorized_keys
+              chmod 0600 ~/.ssh/authorized_keys
             fi

             rm -f '#{remote_path}'
...
```

**Comment by stephencooke** [comment]
is there an ETA on a TAGGED release for this bug and can we get 1.8.5 removed as the main download or more people will be getting this issue.

**Comment by chrisroberts** [comment]
Hi! Thanks for the report of this issue. It was resolved via #7639. Cheers!
